### PR TITLE
Restrictions: chosen options controller

### PR DESCRIPTION
**Pull request by Anlord241**
image
Also check if a test contains questions you answer

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java [hunk 1]
```diff
@@ -33,16 +39,22 @@ public ChosenOption getByTestAndQuestionId(Long testId, Long questionId) {
...
+            test = item.getId().getTest();
+            question = item.getId().getQuestion();
+
+            restrictionsService.checkOwnerIsCurrentUser(test);
+
+            restrictionsService.checkStartedStatus(test);
+
+            restrictionsService.checkTestContainsQuestion(test, question);
         }
-    }
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java:42]: `chosenOptionRepository.saveAll(chosenOptions);`?

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java (hunk 1 above)
Comment (Anlord241) [reply src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java:42]: > `chosenOptionRepository.saveAll(chosenOptions);`?

Full method: 
image

File: src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java [hunk 2]
```diff
@@ -33,16 +39,22 @@ public ChosenOption getByTestAndQuestionId(Long testId, Long questionId) {
...
     @Override
-    public void save(ChosenOption chosenOption) {
-        try {
-            chosenOptionRepository.save(chosenOption);
-        } catch (EntityNotFoundException exception) {
-            throw new ChosenOptionBadRequestException();
+    public void saveAll(List<ChosenOption> chosenOptions) {
+        Test test;
+
+        Question question;
```
Comment (maxim12321) [inline src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java]: Why did you split declaration and initialization?

File: src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java (hunk 1 above)
Comment (AntonMaksimenkau) [reply src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java:42]: And what do you think about parallel stream here?

**Review by maxim12321 — APPROVED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

**Pull request merged**